### PR TITLE
HPCC-13192 Add a more sensible authorization message for older versions ...

### PR DESCRIPTION
**Pull request by jamienoss**
...or redis

Signed-off-by: James Noss james.noss@lexisnexis.com

### CODE REVIEW

**Comment by jamienoss** [comment]
@richardkchapman, @ghalliday please review.

File: plugins/redis/redis.cpp [hunk 1]
```diff
@@ -402,7 +402,7 @@ void Connection::assertOnCommandError(const redisReply * reply, const char * cmd
 }
 void Connection::assertAuthorization(const redisReply * reply)
 {
-    if (strncmp(reply->str, "NOAUTH", 6) == 0)
+    if (reply && reply->str && ( strncmp(reply->str, "NOAUTH", 6) == 0 || strncmp(reply->str, "ERR operation not permitted", 30) == 0 ))
```
Comment (richardkchapman) [inline plugins/redis/redis.cpp]: I count 27 chars not 30...

**Comment by jamienoss** [comment]
@richardkchapman fixed.

File: plugins/redis/redis.cpp [hunk 2]
```diff
@@ -402,7 +402,7 @@ void Connection::assertOnCommandError(const redisReply * reply, const char * cmd
 }
 void Connection::assertAuthorization(const redisReply * reply)
 {
-    if (strncmp(reply->str, "NOAUTH", 6) == 0)
+    if (reply && reply->str && ( strncmp(reply->str, "NOAUTH", 6) == 0 || strncmp(reply->str, "ERR operation not permitted", 27) == 0 ))
```
Comment (ghalliday) [inline plugins/redis/redis.cpp]: future: could use startsWith() in jstring to avoid explicitly counting.  It might even be more efficient.

**Comment by ghalliday** [comment]
@richardkchapman looks good to take.

File: plugins/redis/redis.cpp (hunk 2 above)
Comment (jamienoss) [inline plugins/redis/redis.cpp]: ok thanks, I'll try to remember that.

**Pull request merged**